sidecar/internal/csiaddonsnode/csiaddonsnode.go

```diff
@@ -0,0 +1,130 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csiaddonsnode
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	podNameEnvKey      = "POD_NAME"
+	podNamespaceEnvKey = "POD_NAMESPACE"
+	podUIDEnvKey       = "POD_UID"
+	podIPEnvKey        = "POD_IP"
+)
+
+// Deploy creates CSIAddonsNode custom resource with all required information.
+func Deploy(config *rest.Config, driverName, nodeID, endpoint string) error {
+	object, err := getCSIAddonsNode(driverName, endpoint, nodeID)
+	if err != nil {
+		return err
+	}
+
+	err = csiaddonsv1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return fmt.Errorf("failed to add scheme: %w", err)
+	}
+
+	crdConfig := *config
+	crdConfig.ContentConfig.GroupVersion = &csiaddonsv1alpha1.GroupVersion
+	crdConfig.APIPath = "/apis"
+	crdConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
+	crdConfig.UserAgent = rest.DefaultKubernetesUserAgent()
+
+	c, err := rest.UnversionedRESTClientFor(&crdConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get REST Client: %w", err)
+	}
+
+	newObject := &csiaddonsv1alpha1.CSIAddonsNode{}
+
+	err = c.Post().
+		Resource("csiaddonsnodes").
+		Namespace(object.Namespace).
+		Name(object.Name).
+		Body(object).
+		Do(context.TODO()).
+		Into(newObject)
+
+	if err != nil {
+		return fmt.Errorf("failed to create csiaddonsnode object: %w", err)
+	}
+
+	return nil
+}
+
+// lookupEnc returns environmental variable value given the name.
+func lookupEnv(name string) (string, error) {
+	val, ok := os.LookupEnv(name)
+	if !ok {
+		return val, fmt.Errorf("Required environemental variable %q not found", name)
+	}
+
+	return val, nil
+}
+
+// getCSIAddonsNode fills required information and return CSIAddonsNode object.
+func getCSIAddonsNode(driverName, endpoint, nodeID string) (*csiaddonsv1alpha1.CSIAddonsNode, error) {
+	podName, err := lookupEnv(podNameEnvKey)
+	if err != nil {
+		return nil, err
+	}
+	podNamespace, err := lookupEnv(podNamespaceEnvKey)
+	if err != nil {
+		return nil, err
+	}
+	podUID, err := lookupEnv(podUIDEnvKey)
+	if err != nil {
+		return nil, err
+	}
+	podIP, err := lookupEnv(podIPEnvKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &csiaddonsv1alpha1.CSIAddonsNode{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      podUID,
+			Namespace: podNamespace,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       podName,
+					UID:        types.UID(podUID),
+				},
+			},
+		},
+		Spec: csiaddonsv1alpha1.CSIAddonsNodeSpec{
+			Driver: csiaddonsv1alpha1.CSIAddonsNodeDriver{
+				Name:     driverName,
+				EndPoint: podIP + endpoint,
```

Comment:
do we have to check to make sure the endpoint contains `:` or else this will be a problem?